lib/api/2.0/notification.js

```diff
@@ -30,7 +36,124 @@ var notificationProgressPost = controller(function(req, res){
+ * @apiDescription Receives Redfish alerts, refactors the message and publishs it into amqp
+ * @apiName notification post
+ * @apiGroup notification
+ */
+
+var notificationAlertsPost = controller({success: 201}, function(req,res) {
+    /* The alert would look like the following:
+     {
+     "type": "node",
+     "action": "alerts",
+     "typeId": "58d94cec316779d4126be134",
+     "data": {
+         "Context": "context string",
+         "EventId": "8689",
+         "EventTimestamp": "2017-03-29T15:39:34-0500",
+         "EventType": "Alert",
+         "MemberId": "7e675c8e-127a-11e7-9fc8-64006ac35232",
+         "Message": "The coin cell battery in CMC 1 is not working.",
+         "MessageArgs": ["1"],
+         "MessageArgs@odata.count": 1,
+         "MessageId": "CMC8572",
+         "Severity": "Critical",
+         "sourceIpAddress": "10.240.19.130",
+         "rackhdId": "58d94cec316779d4126be134",
+         "ChassisName": "PowerEdge R630",
+         "ServiceTag": "4666482",
+         "SN": "CN747515A80855"
+         },
+     "severity": "critical",
+     "version": "1.0",
+     "createdAt": "2017-03-29T19:44:12.972Z"
+     }
+     */
+
+    return Promise.try(function(){
+        var ip = req.headers['x-forwarded-for'] ||
+            req.connection.remoteAddress ||
+            req.socket.remoteAddress ||
+            req.connection.socket.remoteAddress;
+        var amqpMessage = {};
+        var deviceInfoCatalog ={};
+        var bmcCatalog = {};
+        var alert = {};
+
+        return new Promise(function(resolve, reject) {
+            var parse = parser.urlencoded({ extended: false });
+            parse(req, res, function (err) {
+                if (err) {
+                    return reject(err);
+                }
+                    resolve();
+            });
+        })
+            .then(function() {
+            return waterline.catalogs.findOne({source: "bmc", "data.IP Address": ip})
+                .then(function(data) {
+                    if (data !== undefined && data.hasOwnProperty("node") && data.node !== ""){
+                         bmcCatalog = data;
+                        //Format the amqp message by adopting the rackHD notification standard
+                        amqpMessage.type = "node";
+                        amqpMessage.action = "alerts";
+                        amqpMessage.typeId = bmcCatalog.node;
+
+                        var httpReq = _.defaults(req.swagger.query || {},
+                            req.query || {},
+                            req.body || {});
+                        if(Object.keys(httpReq).length === 1 ) {
+                            var key = Object.keys(httpReq)[0];
+                            alert = JSON.parse(key);
+                            amqpMessage.data = alert;
+                            amqpMessage.data.sourceIpAddress = ip;
+                            return waterline.catalogs.findOne({source: "ipmi-fru"})
```

Comment:
There may be multiple data for the same catalog source, the most recent one should be the most correct one. please reference https://github.com/RackHD/on-tasks/blob/master/lib/jobs/get-catalog-values-job.js#L68